packages/graphql-server/src/plugins/useRedwoodRealtime.ts

```diff
@@ -19,35 +19,162 @@ export const liveDirectiveTypeDefs = print(
   astFromDirective(GraphQLLiveDirective)
 )
 
+export type LiveQueryStorageMechanism =
+  | RedisLiveQueryStore
+  | InMemoryLiveQueryStore
+
 export type RedwoodRealtimeOptions = {
-  liveQueries?: UseLiveQueryOptions
-  /**
-   * @description Subscriptions passed from the glob import:
-   * import subscriptions from 'src/subscriptions/**\/*.{js,ts}'
-   */
+  liveQueries?: {
+    store:
+      | 'in-memory'
+      | {
+          redis: {
+            channel?: string
+            publishClient: LiveQueryRedisClient
+            subscribeClient: LiveQueryRedisClient
+          }
+        }
+  }
   subscriptions?: {
+    /**
+     * @description Subscriptions passed from the glob import:
+     * import subscriptions from 'src/subscriptions/**\/*.{js,ts}'
+     */
+    store:
+      | 'in-memory'
+      | {
+          redis: {
+            publishClient: LiveQueryRedisClient
+            subscribeClient: LiveQueryRedisClient
+          }
+        }
     subscriptions: SubscriptionGlobImports
-    pubSub: ReturnType<typeof createPubSub>
   }
 }
 
-export const useRedwoodRealtime = (options: RedwoodRealtimeOptions): Plugin => {
-  if (options.liveQueries?.liveQueryStore) {
-    const liveQueryPlugin = useLiveQuery({
-      liveQueryStore: options.liveQueries.liveQueryStore,
+export interface LiveQueryRedisClient {
```

Comment:
This a more generic Redis interface so a dev can have either IORedis or redis clients since they have different type.

The LiveQueryRedisClient used just has to support and implement these methods.

Note  Upstash Redis for the moment doesn't seem to support on and subscribe. Have to confirm,